rust/pypasswords/src/lib.rs

```
@@ -7,57 +7,147 @@
     }
 
     #[cfg(feature = "cbc")]
     #[pyfunction]
     /// Encrypt the data with CBC `TripleDES`
     pub(crate) fn cbc_encrypt(password: &str, data: &str) -> PyResult<String> {
         let result_bytes = passwords::cbc_encrypt(password.as_bytes(), data.as_bytes())
-            .map_err(|err| PyRuntimeError::new_err(err.to_string()))?;
-        String::from_utf8(result_bytes)
-            .map_err(|_err| PyRuntimeError::new_err("Base64 output contained invalid UTF-8"))
+            .map_err(ToPythonError::to_python_error)?;
+        String::from_utf8(result_bytes).map_err(|_| {
```

Clippy (1.96.0)

[error] `map_err(|_|...` wildcard pattern discards the original error

(error)